### PR TITLE
replace travis with azure pipelines

### PR DESCRIPTION
**Pull request by gzussa**

### CODE REVIEW

File: .azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- task: Maven@3
+  inputs:
+    mavenPomFile: 'pom.xml'
+    mavenOptions: '-Xmx3072m'
+    javaHomeOption: 'JDKVersion'
+    jdkVersionOption: '1.8'
+    jdkArchitectureOption: 'x64'
+    publishJUnitResults: true
+    testResultsFiles: '**/surefire-reports/TEST-*.xml'
+    goals: 'package'
```
Comment (nmuesch) [inline .azure-pipelines/azure-pipelines.yml]: should this be `test`?

File: .azure-pipelines/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    #mavenDirectory: # Required when mavenVersionOption == Path
+    #mavenSetM2Home: false # Required when mavenVersionOption == Path
+    #mavenOptions: '-Xmx1024m' # Optional
+    #mavenAuthenticateFeed: false
+    #effectivePomSkip: false
+    #sonarQubeRunAnalysis: false
+    #sqMavenPluginVersionChoice: 'latest' # Required when sonarQubeRunAnalysis == True# Options: latest, pom
+    #checkStyleRunAnalysis: false # Optional
+    #pmdRunAnalysis: false # Optional
+    #findBugsRunAnalysis: false # Optional
```
Comment (nmuesch) [inline .azure-pipelines/azure-pipelines.yml]: Can we remove the commented out options and add them later if needed?

File: .azure-pipelines/azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+steps:
+- task: Maven@3
+  inputs:
+    mavenPomFile: 'pom.xml'
+    mavenOptions: '-Xmx3072m'
+    javaHomeOption: 'JDKVersion'
+    jdkVersionOption: '1.8'
+    jdkArchitectureOption: 'x64'
+    publishJUnitResults: true
```
Comment (nmuesch) [inline .azure-pipelines/azure-pipelines.yml]: whats this 😄

Comment (gzussa) [reply .azure-pipelines/azure-pipelines.yml]: see the `Publish to Azure Pipelines` item in https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/maven?view=azure-devops

File: .azure-pipelines/azure-pipelines.yml (hunk 1 above)
Comment (gzussa) [reply .azure-pipelines/azure-pipelines.yml]: I don't think it does any harm to test the entire packaging.

Comment (nmuesch) [reply .azure-pipelines/azure-pipelines.yml]: Will this run the tests though? From the maven docs it looks like `package` just takes the compiled code and tries to bundle it into (in our case) a jar - http://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html

`mvn test` should handle both the compilation + testing step for us.

Comment (nmuesch) [reply .azure-pipelines/azure-pipelines.yml]: Ahh I see that in the azure-pipelines run that tests did indeed get run, the output is a bit difficult to read, but in that case I don't have strong opinions about the `goal` definition

**Pull request merged**